recipe/build.sh

```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
 
+# ipopt looks only for its own ASL & Mumps now
+curl -fSsL https://github.com/coin-or-tools/ThirdParty-ASL/archive/releases/1.4.3.tar.gz | tar xz && cd ThirdParty-ASL-releases-1.4.3 && ./get.ASL && ./configure --prefix=${PREFIX} && make -j${CPU_COUNT} && make install && cd -
+
+curl -fSsL https://github.com/coin-or-tools/ThirdParty-Metis/archive/releases/1.3.9.tar.gz | tar xz && cd ThirdParty-Metis-releases-1.3.9 && ./get.Metis && ./configure --prefix=${PREFIX} && make -j${CPU_COUNT} && make install && cd -
+
+curl -fSsL https://github.com/coin-or-tools/ThirdParty-Mumps/archive/releases/1.6.2.tar.gz | tar xz && cd ThirdParty-Mumps-releases-1.6.2 && patch -p1 -i ${RECIPE_DIR}/thirdparty-mumps-linux-configure.patch && ./get.Mumps && ./configure --prefix=${PREFIX} && make -j${CPU_COUNT} && make install && cd -
+
```

Comment:
I don't think this is the approach we want to do here.

Reply:
Or, why would we not want to build using the conda-forge built dependencies?

Reply:
I read the first statement in the pull request. Why does building with the external dependencies cause segfaults? And why is this the right approach? The approach in conda & conda-forge seems to favor separating dependencies into separate packages.

Reply:
I did have time to debug further bug it sefault with other versions of mumps.
Also coin carries its own set of patches atop mumps.

Reply:
@richardotis Do you think this is the same segfaults you were seeing with mumps?

Reply:
I guess if IPOPT patches these libraries in the install process, then maybe this is the only option. There's just been an awful lot of work trying to setup a proper dependency chain in conda forge and this abandons it.

Reply:
Sorry to hear that, I guess they didnt follow where upstream was going.

Reply:
@moorepants I wasn't seeing segfaults but I was seeing stochastic test failures which I suspected being related to precision loss somewhere.

As someone who did a lot of work trying to get the dependency chain on Windows working for ipopt, I endorse this approach. We're still building from source here, which is better than just repackaging binaries. IMO, conda-forge doesn't have the resources to deviate too much from upstream ipopt on this, and if getting a working ipopt requires a lot of dependency patching, we would effectively end up maintaining multiple versions of mumps anyway. As long as this version of ipopt is modified not to detect the conda-forge packages, this should be fine.

@jschueller Is it a lot of extra work to get this working for Windows? If you turn the `skip` switch off in the recipe and rerender, does it Just Work?

Reply:
I think these curl calls can be moved into the source section of the meta.yaml.

Reply:
I agree, that if IPOPT is patching all of its dependencies, then we have to follow suit and bundle the dependencies here. We can't realistically maintain special versions of the dependencies just for IPOPT on conda forge. Too bad IPOPT isn't upstreaming its changes to these packages.